code/modules/mob/living/carbon/human/species_types/mothmen.dm

```diff
@@ -47,3 +55,125 @@
 	if(istype(weapon, /obj/item/melee/flyswatter))
 		return 9 //flyswatters deal 10x damage to moths
 	return 0
+
+/datum/species/moth/on_species_gain(mob/living/carbon/human/H)
+	..()
+	H.dna.original_wings = "[H.dna.features["moth_wings"]]" //This stores the current wing type
+	cocoon = new()
+	cocoon.Grant(H)
```

Comment:
you should only grant the action when their wings are burnt

Reply:
Actually I've seen a couple of 'actions' that are always available but are blacked out and can only be used during special circumstances. Do you mean something like that?

Reply:
Nope, the action should not even show up unless the wings are burnt.